cmd/ssh_key_delete.go

```diff
@@ -34,24 +32,29 @@ func (c *computeSSHKeyDeleteCmd) cmdPreRun(cmd *cobra.Command, args []string) er
 }
 
 func (c *computeSSHKeyDeleteCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	ctx := exoapi.WithEndpoint(
-		gContext,
-		exoapi.NewReqEndpoint(account.CurrentAccount.Environment, account.CurrentAccount.DefaultZone),
-	)
+
+	ctx := gContext
 
 	if !c.Force {
 		if !askQuestion(fmt.Sprintf("Are you sure you want to delete SSH key %s?", c.Name)) {
 			return nil
 		}
 	}
 
-	var err error
-	decorateAsyncOperation(fmt.Sprintf("Deleting SSH key %s...", c.Name), func() {
-		err = globalstate.EgoscaleClient.DeleteSSHKey(ctx, account.CurrentAccount.DefaultZone, &egoscale.SSHKey{Name: &c.Name})
+	decorateAsyncOperations(fmt.Sprintf("Deleting SSH key %s...", c.Name), func() error {
+		op, err := globalstate.EgoscaleV3Client.DeleteSSHKey(ctx, c.Name)
+
+		if err != nil {
+			return err
+		}
+
+		_, err = globalstate.EgoscaleV3Client.Wait(ctx, op, v3.OperationStateSuccess)
+		if err != nil {
+			return err
+		}
+
+		return nil
 	})
```

Comment:
You need to check the returned error if you are using `func decorateAsyncOperations(message string, fns ...func() error) error`

```suggestion
	return decorateAsyncOperations(fmt.Sprintf("Deleting SSH key %s...", c.Name), func() error {
		op, err := globalstate.EgoscaleV3Client.DeleteSSHKey(ctx, c.Name)
		if err != nil {
			return err
		}

		_, err = globalstate.EgoscaleV3Client.Wait(ctx, op, v3.OperationStateSuccess)
		return err
	})
```